### PR TITLE
Extend promotions page with order promotions

### PR DESCRIPTION
**Pull request by IKarbowiak**
Documentation for https://github.com/saleor/saleor/pull/14696

### CODE REVIEW

File: docs/developer/discounts/promotions.mdx [hunk 1]
```diff
@@ -162,6 +209,54 @@ Below, you can find examples of promotion rules with catalogue predicates that d
...
+```
+
+- **$5** discount applied when the total price is above **$50** applied on subtotal price
+
+```json
+{
+  "promotionRule": {
+    "orderPredicate": {
+      "discountedObjectPredicate": {
+        "baseTotalPrice": {
```
Comment (korycins) [inline docs/developer/discounts/promotions.mdx:248]: btw. Shouldn't be `baseSubtotalPrice`?

Comment (IKarbowiak) [reply docs/developer/discounts/promotions.mdx:248]: No, why?
The case is:
`**$5** discount applied when the total price is above **$50** applied on subtotal price`

**Review by SzymJ — APPROVED** [review]

**Review by korycins — APPROVED** [review]

File: docs/developer/discounts/promotions.mdx [hunk 2]
```diff
@@ -32,32 +45,75 @@ price will not be instantly visible after promotion creation.
...
 Predicates define the conditions to be met to apply the reward.
 Allows defining complex nested conditions in the same way as in `where` filters, with `AND` and `OR` operators,
 combined with operations (e.g. `eq`, `oneOf`).
 To get to know how to build complex conditions reach out to the [filters page](/api-usage/filtering.mdx#general-approach).
 
-Currently only predicates on catalogue levels are available,
-that allow to definition of the conditions for variants, products, collections and categories.
+### Catalogue predicate
+
+Catalogue level predicates allow to definition of the conditions for variants, products, collections and categories.
```
Comment (zedzior) [inline docs/developer/discounts/promotions.mdx:115]:
```suggestion
Catalogue level predicates allow to define conditions for variants, products, collections and categories.
```

File: docs/developer/discounts/promotions.mdx [hunk 3]
```diff
@@ -254,17 +362,24 @@ To add a promotion rule to the promotion use [`promotionRuleCreate`](../../api-r
...
-
-The following example demonstrates the creation of a promotion rule that applies
-a 50% discount on all products from a collection and categories with specified IDs.
+- `name`: optional name of the promotion rule.
+- `description`: optional description of the promotion rule.
+- `promotion`: define to which promotion the rule should be part of.
+- `channels`: define in which channels the rule should be applicable.
+- `rewardValueType`: define the type of discount `PERCENTAGE`/`FIXED`.
+- `rewardValue`: define the value of the discount.
+- `rewardType`: define the type of reward. Required for `checkoutAndOrder` promotion rules.
```
Comment (zedzior) [inline docs/developer/discounts/promotions.mdx:371]:
```suggestion
- `rewardType`: define the type of reward. Required for order promotion rules.
```

File: docs/developer/discounts/promotions.mdx [hunk 4]
```diff
@@ -362,8 +480,65 @@ As a result, we get a newly created promotion:
 }
 ```
 
+#### Creation of order promotion rule
+
+The following example demonstrates the creation of a order promotion rule that
+gives **10%** subtotal price discount on checkout/draft order that subtotal price is greatest
```
Comment (zedzior) [inline docs/developer/discounts/promotions.mdx:486]:
```suggestion
gives **10%** subtotal price discount on checkout/draft order that subtotal price is greater
```

File: docs/developer/discounts/promotions.mdx [hunk 5]
```diff
@@ -362,8 +480,65 @@ As a result, we get a newly created promotion:
...
+        }
+      }
+    }
+  }
+}
+```
+
 ## Discounted price recalculations
 
+Discounted price are calculating based on catalogue promotions.
```
Comment (zedzior) [inline docs/developer/discounts/promotions.mdx:540]:
```suggestion
Discounted price is calculated based on catalogue promotions.
```

File: docs/developer/discounts/promotions.mdx [hunk 6]
```diff
@@ -8,9 +8,22 @@ sidebar_label: Promotions
 Promotions enable price reductions based on specific conditions.
 These conditions are defined as predicates in the promotion rules.
 A promotion can consist of multiple rules, but each rule can only belong to one promotion.
+
+There are two type od promotions:
+
+- catalogue promotions - reduce the product price visible by the customer,
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply docs/developer/discounts/promotions.mdx:14]: It's not a mistake. It could end with nothing or `,` or `;`

**Review by zedzior — APPROVED** [review]

File: docs/developer/discounts/promotions.mdx [hunk 7]
```diff
@@ -162,6 +218,54 @@ Below, you can find examples of promotion rules with catalogue predicates that d
 }
 ```
 
+### Order predicate
+
+The order predicates enable the definition of conditions for checkout and draft order objects based on their attributes.
+Currently, it is only possible to define conditions based on the base total and base subtotal price.
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply docs/developer/discounts/promotions.mdx:232]: I added the `base prices` explanation to `Glossary`

File: docs/developer/discounts/promotions.mdx [hunk 8]
```diff
@@ -8,9 +8,22 @@ sidebar_label: Promotions
 Promotions enable price reductions based on specific conditions.
 These conditions are defined as predicates in the promotion rules.
 A promotion can consist of multiple rules, but each rule can only belong to one promotion.
+
+There are two type od promotions:
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:12]:
```suggestion
There are two type of promotions:
```

File: docs/developer/discounts/promotions.mdx [hunk 9]
```diff
@@ -8,9 +8,22 @@ sidebar_label: Promotions
...
+
+- catalogue promotions - reduce the product price visible by the customer,
+- order promotions - applied on checkout / draft order when object fulfill
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:15]:
```suggestion
- order promotions - applied on checkout / draft order when the object fulfill
```

File: docs/developer/discounts/promotions.mdx [hunk 10]
```diff
@@ -8,9 +8,22 @@ sidebar_label: Promotions
...
+  the rule conditions.
+
+All rules within promotion must be of one type `CATALOGUE` or `ORDER`.
+The type of rule is determine by the promotion to which the rule is assigned.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:19]:
```suggestion
The type of rule is determined by the promotion to which the rule is assigned.
```

File: docs/developer/discounts/promotions.mdx [hunk 11]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
 - Define the product fixed or parentage discounts based on certain conditions.
 - Define the discount available only for a specific period.
 
+### Order promotions
+
+The discounts from order rules are automatically applied during checkout or draft order when certain conditions are met.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:50]:
```suggestion
The discounts from order rules are automatically applied during checkout or draft orders when certain conditions are met.
```

File: docs/developer/discounts/promotions.mdx [hunk 12]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
+Only the discount from the promotion rule that provides the maximum savings for the customer is applied.
+
+The order discounts can be combined with catalogue discount, and in such cases, the discount is calculated based on the discounted price.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:53]:
```suggestion
The order discounts can be combined with catalogue discounts, and in such cases, the discount is calculated based on the discounted price.
```

File: docs/developer/discounts/promotions.mdx [hunk 13]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
+
+This kind of promotion discount required providing the `rewardType` that described what
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:55]:
```suggestion
This kind of promotion discount requires providing the `rewardType` that describes what
```

File: docs/developer/discounts/promotions.mdx [hunk 14]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
+kind of discount will be applied after meeting the conditions.
+
+Current available reward types are:
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:58]:
```suggestion
Currently available reward types are:
```

File: docs/developer/discounts/promotions.mdx [hunk 15]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
+
+- `SUBTOTAL_DISCOUNT` - applies the discount on subtotal price
+
+For example, consider a checkout with only one product that is on catalogue discount, and the discounted price is **$12**.
+The best promotion rule for this checkout is giving **10%** discount on subtotal price.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:63]:
```suggestion
The best promotion rule for this checkout is giving a **10%** discount on the subtotal price.
```

File: docs/developer/discounts/promotions.mdx [hunk 16]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
+In this scenario, the order discount will be **10%** from **$12** resulting in a savings of **$1.2**.
+
+:::note
+For now, the order promotions are applicable only on checkouts, and not on draft orders.
+:::
+
+:::info
+Number od `PromotionRule` with `ORDER` predicate type is limited to 100.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:71]:
```suggestion
The number of `PromotionRule` with `ORDER` predicate type is limited to 100.
```

File: docs/developer/discounts/promotions.mdx [hunk 17]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
+
+The type of promotion determines whether to give the `CATALOGUE` or `ORDER` discount.
+For the `CATALOGUE` type, all rules within the promotion must have the `cataloguePredicate` defined,
+while for the `ORDER` type, the `orderPredicate` must be defined.
 
-## Rewards
+## Rewards value
 
-Rewards are defined on the promotion rule level and specify the discount type and value
+Reward value is defined on the promotion rule level and specify the amount and type of the discount
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:98]:
```suggestion
Reward value is defined on the promotion rule level and specifies the amount and type of the discount
```

File: docs/developer/discounts/promotions.mdx [hunk 18]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
 that will be applied when the defined conditions are met.
 Currently, there is only `FIXED` or `PRECENTAGE` discount available.
 
+## Reward type
+
+Reward type is applicable only for order promotions and specify what kind of reward
+will be given when the defined conditions are met.
+Currently, only `SUBTOTAL_DISCOUNT` is available so that the order subtotal price will be reduced.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:106]:
```suggestion
Currently, only `SUBTOTAL_DISCOUNT` is available so the order subtotal price will be reduced.
```

File: docs/developer/discounts/promotions.mdx [hunk 19]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
+
+## Base prices
+
+Base prices are the prices after applying catalogue promotions and do not include vouchers or order discounts.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:110]:
```suggestion
Base prices are the prices after applying for catalogue promotions and do not include vouchers or order discounts.
```

File: docs/developer/discounts/promotions.mdx [hunk 20]
```diff
@@ -32,32 +45,83 @@ price will not be instantly visible after promotion creation.
...
 Predicates define the conditions to be met to apply the reward.
 Allows defining complex nested conditions in the same way as in `where` filters, with `AND` and `OR` operators,
 combined with operations (e.g. `eq`, `oneOf`).
 To get to know how to build complex conditions reach out to the [filters page](/api-usage/filtering.mdx#general-approach).
 
-Currently only predicates on catalogue levels are available,
-that allow to definition of the conditions for variants, products, collections and categories.
+### Catalogue predicate
+
+Catalogue level predicates allow to define conditions for variants, products, collections and categories.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:123]:
```suggestion
Catalogue level predicates allow defining conditions for variants, products, collections, and categories.
```

File: docs/developer/discounts/promotions.mdx [hunk 21]
```diff
@@ -254,17 +370,24 @@ To add a promotion rule to the promotion use [`promotionRuleCreate`](../../api-r
...
-a 50% discount on all products from a collection and categories with specified IDs.
+- `name`: optional name of the promotion rule.
+- `description`: optional description of the promotion rule.
+- `promotion`: define to which promotion the rule should be part of.
+- `channels`: define in which channels the rule should be applicable.
+- `rewardValueType`: define the type of discount `PERCENTAGE`/`FIXED`.
+- `rewardValue`: define the value of the discount.
+- `rewardType`: define the type of reward. Required for order promotion rules.
+- `cataloguePredicate`: defines the conditions to be met to apply catalogue discount.
+  **Required for rule that belong to `CATALOGUE` promotion.**
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:381]:
```suggestion
  **Required for the rule that belongs to `CATALOGUE` promotion.**
```

File: docs/developer/discounts/promotions.mdx [hunk 22]
```diff
@@ -254,17 +370,24 @@ To add a promotion rule to the promotion use [`promotionRuleCreate`](../../api-r
...
+  Allows defining complex nested conditions with `AND` and `OR` operators.
+- `orderPredicate`: defines the conditions to be met to apply order discount.
+  **Required for rule that belong to `ORDER` promotion.**
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:384]:
```suggestion
  **Required for the rule that belongs to `ORDER` promotion.**
```

File: docs/developer/discounts/promotions.mdx [hunk 23]
```diff
@@ -362,8 +488,65 @@ As a result, we get a newly created promotion:
 }
 ```
 
+#### Creation of order promotion rule
+
+The following example demonstrates the creation of a order promotion rule that
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:493]:
```suggestion
The following example demonstrates the creation of an order promotion rule that
```

File: docs/developer/discounts/promotions.mdx [hunk 24]
```diff
@@ -587,8 +744,355 @@ the `order.lines.undiscountedUnitPrice` and `order.lines.unitPrice`. The sum of
 discounts multiplied by the quantity of the line gives the total discount, which is
 equal to the difference between `order.total` and `order.undiscountedTotal`.
 
+## Processing checkout with an applied order promotion
+
+Below you can see an example of checkout data with applied order discount from order promotion
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:749]:
```suggestion
Below you can see an example of checkout data with an applied order discount from an order promotion
```

File: docs/developer/discounts/promotions.mdx [hunk 25]
```diff
@@ -587,8 +744,355 @@ the `order.lines.undiscountedUnitPrice` and `order.lines.unitPrice`. The sum of
...
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+The discount is visible on the discount.amount field at the checkout level,
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:831]:
```suggestion
The discount is visible on the `discount.amount` field at the checkout level,
```

File: docs/developer/discounts/promotions.mdx [hunk 26]
```diff
@@ -587,8 +744,355 @@ the `order.lines.undiscountedUnitPrice` and `order.lines.unitPrice`. The sum of
...
+      },
+      "errors": []
+    }
+  }
+}
+```
+
+## Catalogue and order promotions
+
+Catalogue and order promotions can be mixed together. In such case the
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:903]:
```suggestion
Catalogue and order promotions can be mixed. In such cases the
```

File: docs/developer/discounts/promotions.mdx [hunk 27]
```diff
@@ -587,8 +744,355 @@ the `order.lines.undiscountedUnitPrice` and `order.lines.unitPrice`. The sum of
...
+order discount is applied on discounted price.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:904]:
```suggestion
order discount is applied on the discounted price.
```

File: docs/developer/discounts/promotions.mdx [hunk 28]
```diff
@@ -587,8 +744,355 @@ the `order.lines.undiscountedUnitPrice` and `order.lines.unitPrice`. The sum of
...
 ## Catalogue Promotion and Voucher
 
 Catalogue promotions and vouchers can be combined. In this case, the voucher discount
 is applied to the price after the promotion discount.
 [Here](developer/discounts/vouchers.mdx/#catalogue-promotion-and-voucher-together) you can find the example of such scenario.
+
+## Order Promotion and Voucher
+
+Order promotion cannot be mixed with Vouchers.
+When voucher code is added, the discount from order promotion is not applied.
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:1074]:
```suggestion
When a voucher code is added, the discount from the order promotion is not applied.
```

File: docs/developer/discounts/promotions.mdx [hunk 29]
```diff
@@ -587,8 +744,355 @@ the `order.lines.undiscountedUnitPrice` and `order.lines.unitPrice`. The sum of
...
+    A(Base undiscounted price) --> B[Apply catalogue discounts]
+    B --> Z{Voucher code added?}
+    Z --> |No| K[Apply `checkoutAndOrder` discount]
+    K --> E
+    Z --> |Yes| C[Apply lines vouchers - `ONCE_PER_ORDER`, `SPECIFIC_PRODUCT`]
+    C --> D[Apply `ENTIRE_ORDER` voucher]
+    D --> E[Apply taxes]
+```
+
+#### Below chart is showing the process of applying `ORDER` promotion discount
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:1091]:
```suggestion
#### Below the chart shows the process of applying the `ORDER` promotion discount
```

File: docs/developer/discounts/promotions.mdx [hunk 30]
```diff
@@ -587,8 +744,355 @@ the `order.lines.undiscountedUnitPrice` and `order.lines.unitPrice`. The sum of
...
+
+```mermaid
+graph TD
+    A(Base discounted price) --> B[Collect all `ORDER` promotion rules that apply]
+    B --> C[Choose the promotion rule hat gives the best saving for user]
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:1096]:
```suggestion
    B --> C[Choose the promotion rule that gives the best saving for the user]
```

File: docs/developer/discounts/promotions.mdx [hunk 31]
```diff
@@ -587,8 +744,355 @@ the `order.lines.undiscountedUnitPrice` and `order.lines.unitPrice`. The sum of
...
 [Here](developer/discounts/vouchers.mdx/#catalogue-promotion-and-voucher-together) you can find the example of such scenario.
+
+## Order Promotion and Voucher
+
+Order promotion cannot be mixed with Vouchers.
+When voucher code is added, the discount from order promotion is not applied.
+
+## Charts that show the discounts flow
+
+#### Following chart is showing an order in which discounts apply:
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:1078]:
```suggestion
#### The following chart shows an order in which discounts apply:
```

**Review by 2can — CHANGES_REQUESTED** [review]

File: docs/developer/discounts/promotions.mdx [hunk 32]
```diff
@@ -8,9 +8,22 @@ sidebar_label: Promotions
 Promotions enable price reductions based on specific conditions.
 These conditions are defined as predicates in the promotion rules.
 A promotion can consist of multiple rules, but each rule can only belong to one promotion.
+
+There are two type of promotions:
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:12]:
```suggestion
There are two types of promotions:
```

File: docs/developer/discounts/promotions.mdx [hunk 33]
```diff
@@ -8,9 +8,22 @@ sidebar_label: Promotions
...
+
+- catalogue promotions - reduce the product price visible by the customer,
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:14]:
```suggestion
- catalogue promotions - reduce the product price visible to the customer,
```

File: docs/developer/discounts/promotions.mdx [hunk 34]
```diff
@@ -8,9 +8,22 @@ sidebar_label: Promotions
...
+- order promotions - applied on checkout / draft order when the object fulfill
```
Comment (2can) [inline docs/developer/discounts/promotions.mdx:15]:
```suggestion
- order promotions - applied on checkout / draft order when the object fulfilled
```

**Review by 2can — CHANGES_REQUESTED** [review]

**Review by 2can — APPROVED** [review]

**Pull request merged**